server/src/header.js

```diff
@@ -0,0 +1,63 @@
+const React = require("react");
+const PropTypes = require("prop-types");
+const { Localized } = require("fluent-react/compat");
+const sendEvent = require("./browser-send-event.js");
+const { SignInButton } = require("./signin-button.js");
+
+exports.Header = class Header extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  onClickMyShots() {
+    sendEvent("goto-myshots", "homepage", {useBeacon: true});
+    window.location = "/shots";
+  }
+
+  renderFxASignIn() {
+    if (!this.props.initiatePage) {
+      return (
+        this.props.isFirefox && this.props.hasDeviceId ?
+          <SignInButton isAuthenticated={this.props.hasFxa} initiatePage={this.props.initiatePage} /> : null
+      );
+    }
+
+    return (
+      this.props.enableUserSettings && this.props.hasDeviceId ?
+        <SignInButton isAuthenticated={this.props.hasFxa} initiatePage="shots" /> : null
+    );
+  }
+
+  render() {
+    let myShots;
+    if (this.props.hasDeviceId && !this.props.initiatePage) {
+      myShots = <button className="nav-button" onClick={ this.onClickMyShots.bind(this) } tabIndex="0">
+        <div className="icon icon-shots"></div>
+        <Localized id="gMyShots">
+          <span>My Shots</span>
+        </Localized>
+      </button>;
+    }
+
+    const signin = this.renderFxASignIn();
+    const search = this.props.searchForm && this.props.initiatePage === "shots" ? this.props.searchForm : null;
```

Comment:
Just want to +1 on Jared's concern regarding how and where the content of the header is inserted.  We're already special casing for My Shots here; it doesn't feel right.

Reply:
I agree it does seem a bit off to have searchForm moved to Header component when rest of its styles and event handlers are in shotIndex view. Will update to move SearchForm rendering outside of Header component